### PR TITLE
Rework "installation" sections by introducing "self-hosted" section

### PR DESCRIPTION
**Pull request by matkuliak**
## Summary of the changes / Why this is an improvement

Changes that we've agreed on with @proddata 
- Rename the "CrateDB and Linux" -> "Self-hosted CrateDB" which will host installs for various systems
- Remove the Windows mentions from [Ad-hoc install](https://crate.io/docs/crate/tutorials/en/latest/basic/index.html#ad-hoc-unix-macos-windows) and move it to mentioned "Self-hosted CrateDB". Small additions to the tutorial itself, with the installation way that we've agreed on, it didn't need much change.
    - Initial task asked for some screenshots, I don't think they're necessary here, opinions? 

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: docs/self-hosted/windows.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+This section of the documentation outlines how to use the release archives to
+install CrateDB.
+
+.. CAUTION::
+
+    We do not yet officially support CrateDB on Windows for production use. If
+    you would like to deploy CrateDB on Windows, please feel free to `contact
+    us`_ so we can work with you on a solution.
+
+    CrateDB versions before 4.2 required a separate Java installation. For
```
Comment (proddata) [inline docs/self-hosted/windows.rst]: I think we can remove the 4.2 notice by now, as it has been released 2 years ago and isn't really supported anymore. @amotl what do you think?

File: docs/self-hosted/windows.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    CrateDB 3.0 to 4.1, Java 11 is the minimum requirement. CrateDB versions
+    before 3.0 require Java 8. We recommend using `Oracle Java`_ on Microsoft
+    Windows.
+
+#. Download the latest `CrateDB release`_ from our Windows archive.
```
Comment (proddata) [inline docs/self-hosted/windows.rst]:
```suggestion
#. Download the latest `CrateDB release archive`_ for Windows.
```

File: docs/self-hosted/windows.rst [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+will look like this.
+
+.. code-block:: doscon
+
+    PS> ./bin/crate
+
+.. _Powershell: https://docs.microsoft.com/en-us/powershell/scripting/overview?view=powershell-7.2
+.. _starting PowerShell: https://docs.microsoft.com/en-us/powershell/scripting/learn/ps101/01-getting-started?view=powershell-7.1#how-do-i-launch-powershell
+.. _Windows Server: https://www.microsoft.com/en-us/windows-server
+.. _CrateDB release: https://cdn.crate.io/downloads/releases/cratedb/x64_windows/
```
Comment (proddata) [inline docs/self-hosted/windows.rst]:
```suggestion
.. _CrateDB release archive: https://cdn.crate.io/downloads/releases/cratedb/x64_windows/
```

File: docs/self-hosted/windows.rst [hunk 4]
```diff
@@ -0,0 +1,77 @@
+.. _windows-install:
+
+=================
+Microsoft Windows
```
Comment (proddata) [inline docs/self-hosted/windows.rst]:
```suggestion
Windows
```

Maybe just Windows?

File: docs/self-hosted/windows.rst [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+=================
+
+This section of the documentation outlines how to use the release archives to
+install CrateDB.
```
Comment (proddata) [inline docs/self-hosted/windows.rst]:
```suggestion
This section of the documentation outlines how to use the release archives to
run CrateDB on Microsoft Windows.
```

File: docs/self-hosted/windows.rst (hunk 4 above)
Comment (proddata) [reply docs/self-hosted/windows.rst]: We don't mention Canonical  for Ubuntu either

File: docs/self-hosted/windows.rst (hunk 5 above)
Comment (proddata) [reply docs/self-hosted/windows.rst]: in the running text I think it is fine

File: docs/self-hosted/windows.rst [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+       Terminate batch job (Y/N)? Y
+
+.. SEEALSO::
+
+      Consult the :ref:`crate-reference:cli` documentation for further information
+      about the ``./bin/crate`` command.
+
+
+Notes about Microsoft Windows
```
Comment (proddata) [inline docs/self-hosted/windows.rst]: Maybe move that to an admonition block?

**Review by proddata — COMMENTED** [review]
In general looks good.
I think this might need some redirects on the server side

File: docs/self-hosted/windows.rst (hunk 1 above)
Comment (amotl) [reply docs/self-hosted/windows.rst]: I think it is fine to remove the notice.

File: docs/basic/index.rst [hunk 7]
```diff
@@ -319,16 +293,18 @@ Configuration files
 When using the package-based setup flavor for :ref:`install-deb` or
 :ref:`install-rpm`, the main CrateDB configuration files are located within the
 ``/etc/crate`` directory.
-When using the :ref:`install-adhoc` setup, the configuration files are located
-within the ``config/`` directory.
+When using the :ref:`install-adhoc` setup or the 
+:ref:`Microsoft Windows <windows-install>` setup the configuration files are
```
Comment (amotl) [inline docs/basic/index.rst]:
```suggestion
:ref:`Microsoft Windows <windows-install>` setup, the configuration files are
```

File: docs/self-hosted/windows.rst [hunk 8]
```diff
@@ -0,0 +1,76 @@
+.. _windows-install:
+
+=================
+Windows
+=================
```
Comment (amotl) [inline docs/self-hosted/windows.rst]: reStructuredText requires the headline armor to be aligned with the length of the section title.

```suggestion
=======
Windows
=======
```

File: docs/self-hosted/windows.rst [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+.. _Powershell: https://docs.microsoft.com/en-us/powershell/scripting/overview?view=powershell-7.2
+.. _starting PowerShell: https://docs.microsoft.com/en-us/powershell/scripting/learn/ps101/01-getting-started?view=powershell-7.1#how-do-i-launch-powershell
+.. _Windows Server: https://www.microsoft.com/en-us/windows-server
+.. _CrateDB release archive: https://cdn.crate.io/downloads/releases/cratedb/x64_windows/
+.. _msvcrt ARM64: https://aka.ms/vs/16/release/VC_redist.arm64.exe
+.. _msvcrt x86-32: https://aka.ms/vs/16/release/vc_redist.x86.exe
+.. _msvcrt x86-64: https://aka.ms/vs/16/release/vc_redist.x64.exe
+.. _7-Zip: https://www.7-zip.org/
+.. _Oracle Java: https://www.oracle.com/java/technologies/javase-downloads.html
+.. _contact us: https://crate.io/contact/
```
Comment (amotl) [inline docs/self-hosted/windows.rst]: We employ a convention to sort those references alphanumerically. Can I ask you to change their order accordingly?

**Review by amotl — CHANGES_REQUESTED** [review]
Thank you, @matkuliak. I've added some inline comments to this patch.

@proddata:

> I think this might need some redirects on the server side.

Essentially, we need a single redirect from `/linux/*` to `/self-hosted/*`. Right?

**Review by amotl — APPROVED** [review]
Thank you. Note: Please don't merge/rebuild until the redirect is in place.

**Comment by amotl** [comment]
The redirect is in place, so it is good to go.

**Pull request merged**